docs/sources/developers/plugins/migration-guide.md

```diff
@@ -4,77 +4,451 @@ title = "Plugin migration guide"
+  })
+}
+
+// after
+package main
+
+import (
+  "os"
+
+  "github.com/grafana/grafana-plugin-sdk-go/backend/log"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+
+  "github.com/myorgid/datasource/pkg/plugin"
+)
+
+func main() {
+  log.DefaultLogger.Debug("Running GRPC server")
+
+  if err := datasource.Manage("myorgid-datasource", NewSampleDatasource, datasource.ManageOpts{}); err != nil {
+				log.DefaultLogger.Error(err.Error())
+				os.Exit(1)
+		}
+}
+```
+
+#### 3. Update the plugin package
+
+Update your `plugin` package to use the new plugin sdk.
+
+```go
+// before
+package plugin
+
+import (
+  "context"
+
+  "github.com/grafana/grafana_plugin_model/go/datasource"
+  "github.com/hashicorp/go-hclog"
+)
+
+func NewSampleDatasource(pluginLogger hclog.Logger) (*SampleDatasource, error) {
+	return &SampleDatasource{
+		logger: pluginLogger,
+	}, nil
+}
+
+type SampleDatasource struct{
+  logger hclog.Logger
+}
+
+func (d *SampleDatasource) Query(ctx context.Context, tsdbReq *datasource.DatasourceRequest) (*datasource.DatasourceResponse, error) {
+  d.logger.Info("QueryData called", "request", req)
+  // logic for querying your datasource.
+}
+
+// after
+package plugin
+
+import (
+  "context"
+
+  "github.com/grafana/grafana-plugin-sdk-go/backend"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/log"
+  "github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+func NewSampleDatasource(_ backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	return &SampleDatasource{}, nil
+}
+
+type SampleDatasource struct{}
+
+
+func (d *SampleDatasource) Dispose() {
+	// Clean up datasource instance resources.
+}
+
+func (d *SampleDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+  log.DefaultLogger.Info("QueryData called", "request", req)
+  // logic for querying your datasource.
+}
+
+func (d *SampleDatasource) CheckHealth(_ context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+  log.DefaultLogger.Info("CheckHealth called", "request", req)
+  // The main use case for these health checks is the test button on the
+  // datasource configuration page which allows users to verify that
+  // a datasource is working as expected.
+}
+```
+
+### Unsigned backend plugins will not be loaded
+
+We strongly recommend our Grafana users not to allow unsigned plugins in their Grafana installation. By allowing unsigned plugins, we can’t guarantee the authenticity of the plugin which could compromise the security of your Grafana installation.
+
+This means that you, as a plugin developer, need to get your plugin signed if you want it to be able to run on all Grafana installations.
+
+Follow the following [steps](https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/#sign-a-plugin) to get instructions on how to set up plugin signing.
```

Comment:
```suggestion
To sign your plugin, see [Sign a plugin](https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/#sign-a-plugin).
```

Reply:
Thanks @osg-grafana, although it seems that this was already changed based on @achatterjee-grafana's suggestion before: https://github.com/grafana/grafana/pull/38911#discussion_r705557323

Reply:
One of us can always go in later and do a second review. All good. "Follow the following ..." seems bloated.

Reply:
Applied your suggestion in https://github.com/grafana/grafana/pull/38911/commits/e11fa23f25d26ce162ddb4432fe2c6d909f01176.